### PR TITLE
Fix fanout in customers model

### LINKED ISSUE #1: Customers mart has a fan-out when an order contains multiple items
An example is customer `6521e8e2-e417-49c0-a9e0-901fc6ea2c31`, where lifetime_spend_pretax is $17 but lifetime_spend is $35.36:
image

This happens because of these lines: 
https://github.com/dbt-labs/jaffle-shop/blob/15f0b978fb1c08173f37c7cfecb5c49efcd4bba6/models/marts/customers.sql#L29-L36

lifetime_spend_pretax comes from adding these two rows together:
image

But the order's tax-inclusive cost is rolled up, so a fanout happens and $17.68 is doubled:
image

### PR DESCRIPTION
**Pull request by gwenwindflower**
Closes #1 

This turned into pulling a sweater thread that unravelled some issues in the source data, so I had to dig in to quite a bit!

### CODE REVIEW

File: Taskfile.yml
Comment (gwenwindflower) [inline Taskfile.yml:1]: We expect Cloud CLI to be installed by `brew` so no venv activation needed for these

File: dbt_project.yml
Comment (gwenwindflower) [inline dbt_project.yml:1]: This is the version sqlfluff installs by default in the pre-commit hook environment via sqlfluff-dbt-templater so we need to allow for it, although I anticipated primarily dbt Cloud CLI with versionless as the prime target for this project

File: models/marts/orders.sql [hunk 1]
```diff
@@ -6,7 +6,7 @@ orders as (
 
 ),
 
-order_items_table as (
+order_items as (
```
Comment (gwenwindflower) [inline models/marts/orders.sql:9]: we no longer need to do this renaming as this was for a DuckDB issue

File: models/marts/customers.yml [hunk 2]
```diff
@@ -1,6 +1,9 @@
 models:
   - name: customers
     description: Customer overview data mart, offering key details for each unique customer. One row per customer.
+    tests:
+      - dbt_utils.expression_is_true:
+          expression: "lifetime_spend_pretax + lifetime_tax_paid = lifetime_spend"
```
Comment (gwenwindflower) [inline models/marts/customers.yml:6]: This is the core thing this PR exists to make true

**Review by dave-connors-3 — APPROVED** [review]
🦞

**Pull request merged**